ui/components/app/network-display/network-display.js

```diff
@@ -1,129 +1,26 @@
-import { getProviderConfig } from '../../../ducks/metamask/metamask';
-import { getNetworkLabelKey } from '../../../helpers/utils/i18n-helper';
-
-/**
- * @deprecated The `<NetworkDisplay />` component has been deprecated in favor of the new `<PickerNetwork>` component from the component-library.
- * Please update your code to use the new `<PickerNetwork>` component instead, which can be found at ui/components/component-library/picker-network/picker-network.tsx.
- * You can find documentation for the new `PickerNetwork` component in the MetaMask Storybook:
- * {@link https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-pickernetwork--docs}
- * If you would like to help with the replacement of the old `NetworkDisplay` component, please submit a pull request against this GitHub issue:
- * {@link https://github.com/MetaMask/metamask-extension/issues/20485}
- */
-
-export default function NetworkDisplay({
-  indicatorSize,
-  disabled,
-  labelProps,
-  targetNetwork,
-  onClick,
-}) {
-  const networkIsLoading = useSelector(isNetworkLoading);
-  const providerConfig = useSelector(getProviderConfig);
-  const t = useI18nContext();
-
-  const { nickname, type: networkType } = targetNetwork ?? providerConfig;
+import { getCurrentNetwork } from '../../../selectors';
+import { PickerNetwork, AvatarNetworkSize } from '../../component-library';
 
+export default function NetworkDisplay() {
+  const currentNetwork = useSelector(getCurrentNetwork);
   return (
-    <Chip
-      dataTestId="network-display"
-      borderColor={
-        onClick ? BorderColor.borderDefault : BorderColor.borderMuted
-      }
-      onClick={onClick}
-      leftIcon={
-        <LoadingIndicator
-          alt={t('attemptingConnect')}
-          title={t('attemptingConnect')}
-          isLoading={networkIsLoading}
-        >
-          <ColorIndicator
-            color={
-              networkType === NETWORK_TYPES.RPC
-                ? IconColor.iconMuted
-                : networkType
-            }
-            size={indicatorSize}
-            type={ColorIndicator.TYPES.FILLED}
-            iconClassName={
-              networkType === NETWORK_TYPES.RPC && indicatorSize !== Size.XS
-                ? 'fa fa-question'
-                : undefined
-            }
-          />
-        </LoadingIndicator>
-      }
-      rightIcon={
-        onClick ? <Icon name={IconName.ArrowDown} size={IconSize.Xs} /> : null
-      }
-      label={
-        networkType === NETWORK_TYPES.RPC
-          ? nickname ?? t('privateNetwork')
-          : t(getNetworkLabelKey(networkType))
-      }
-      className={classnames('network-display', {
-        'network-display--disabled': disabled,
-        'network-display--clickable': typeof onClick === 'function',
-      })}
-      labelProps={{
-        variant: TypographyVariant.H7,
-        ...labelProps,
-      }}
+    <PickerNetwork
+      className="network-display"
+      label={currentNetwork?.nickname}
```

Comment:
This should fix the failing e2e tests

```suggestion
      label={currentNetwork?.nickname}
      labelProps={{ 'data-testid': 'network-display'}}
```

<img width="1123" alt="Screenshot 2024-06-25 at 7 18 48 PM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/1076f5f7-a3f4-4c6e-8fbb-e12adc0c3a1c">